impl/src/main/java/org/ehcache/internal/store/tiering/CacheStore.java

```diff
@@ -0,0 +1,350 @@
+    try {
+      return authoritativeTier.computeIfPresent(key, remappingFunction);
+    } finally {
+      // XXX: it's not possible to discriminate if computation has been performed or not by looking at the computeIfPresent return value
+      // Hence, we must always clear the caching tier even when that's not necessary (ie: when no computing happened)
+      cachingTier.remove(key);
+    }
+  }
+
+  @Override
+  public ValueHolder<V> computeIfPresent(final K key, final BiFunction<? super K, ? super V, ? extends V> remappingFunction, final NullaryFunction<Boolean> replaceEqual) throws CacheAccessException {
+    try {
+      return authoritativeTier.computeIfPresent(key, remappingFunction, replaceEqual);
+    } finally {
+      // XXX: it's not possible to discriminate if computation has been performed or not by looking at the computeIfPresent return value
+      // Hence, we must always clear the caching tier even when that's not necessary (ie: when no computing happened)
+      cachingTier.remove(key);
+    }
+  }
+
+  @Override
+  public Map<K, ValueHolder<V>> bulkCompute(Set<? extends K> keys, Function<Iterable<? extends Map.Entry<? extends K, ? extends V>>, Iterable<? extends Map.Entry<? extends K, ? extends V>>> remappingFunction) throws CacheAccessException {
+    try {
+      return authoritativeTier.bulkCompute(keys, remappingFunction);
+    } finally {
+      for (K key : keys) {
+        cachingTier.remove(key);
+      }
+    }
+  }
+
+  @Override
+  public Map<K, ValueHolder<V>> bulkCompute(Set<? extends K> keys, Function<Iterable<? extends Map.Entry<? extends K, ? extends V>>, Iterable<? extends Map.Entry<? extends K, ? extends V>>> remappingFunction, NullaryFunction<Boolean> replaceEqual) throws CacheAccessException {
+    try {
+      return authoritativeTier.bulkCompute(keys, remappingFunction, replaceEqual);
+    } finally {
+      for (K key : keys) {
+        cachingTier.remove(key);
+      }
+    }
+  }
+
+  @Override
+  public Map<K, ValueHolder<V>> bulkComputeIfAbsent(Set<? extends K> keys, Function<Iterable<? extends K>, Iterable<? extends Map.Entry<? extends K, ? extends V>>> mappingFunction) throws CacheAccessException {
+    try {
+      return authoritativeTier.bulkComputeIfAbsent(keys, mappingFunction);
+    } finally {
+      for (K key : keys) {
+        cachingTier.remove(key);
+      }
+    }
+  }
+
+  public static class Provider implements Store.Provider {
+
+    private ServiceProvider serviceProvider;
+    private ServiceConfiguration<?> serviceConfiguration;
+    private OnHeapStore.Provider onHeapStoreProvider;
+    private DiskStore.Provider diskStoreProvider;
+
+    @Override
+    public <K, V> Store<K, V> createStore(Configuration<K, V> storeConfig, ServiceConfiguration<?>... serviceConfigs) {
+      //todo use the storeConfig to figure out what providers to use
```

Comment:
We need to make sure that's clearly explained in the associated falloff issue (it may well already be the case, just thought I'd highlight it)